Slight TestBuildEnvUsage2 speedup

aaronlehmann opened the pull request:
TestBuildEnvUsage2 was flagged in https://github.com/docker/docker/issues/19425
as one of the slowest integration tests. It's slow because it has some
comprehensive builder test cases that end up creating a lot of layers.
Even with a busybox base image, this can be expensive. It's not possible
to build "FROM scratch" because the test cases need the shell to ensure
environment variables are set correctly.

Some of the ENV and RUN statements can be combined. This causes fewer
layers to get created. Doing this produces a marginal improvement in the
runtime.

Before:

```
PASS: docker_cli_build_test.go:3956: DockerSuite.TestBuildEnvUsage2 43.619s
```

After:

```
PASS: docker_cli_build_test.go:3956: DockerSuite.TestBuildEnvUsage2 31.286s
```

cc @duglin

Commit:
Slight TestBuildEnvUsage2 speedup

TestBuildEnvUsage2 was flagged in https://github.com/docker/docker/issues/19425
as one of the slowest integration tests. It's slow because it has some
comprehensive builder test cases that end up creating a lot of layers.
Even with a busybox base image, this can be expensive. It's not possible
to build "FROM scratch" because the test cases need the shell to ensure
environment variables are set correctly.

Some of the ENV and RUN statements can be combined. This causes fewer
layers to get created. Doing this produces a marginal improvement in the
runtime.

Before:

    PASS: docker_cli_build_test.go:3956: DockerSuite.TestBuildEnvUsage2 43.619s

After:

    PASS: docker_cli_build_test.go:3956: DockerSuite.TestBuildEnvUsage2 31.286s

Signed-off-by: Aaron Lehmann <aaron.lehmann@docker.com>

calavera commented:
LGTM

duglin commented on integration-cli/docker_cli_build_test.go:
```diff
@@ -4004,30 +3992,15 @@ RUN    [ "$abc" = '"foo"' ]
-ENV    mypath=${mypath:+$mypath:}/away
-RUN    [ "$mypath" = '/home:/away' ]
+ENV    def1=${abc:-DEF} def2=${ccc:-DEF}
+ENV    def3=${ccc:-${def2}xx} def4=${abc:+ALT} def5=${abc:+${abc}:} def6=${ccc:-\$abc:} def7=${ccc:-\${abc}:}
```
Double check me on this one but I think `def5=${abc:+${abc}:}` needs to use something other than "abc" in the 2nd spot otherwise we won't know whether it used the first or section variable. I think `def5=${abc:+${def2}:}` should do it - but then you need to change $def5's result in the next line to "DEF" not "ABC".  

Also, the previous version of that test assigned the result to the same var being used in the expression - I think we've lost that aspect of the test now.

aaronlehmann commented:
Changed that one to `def5=${def2:+${abc}:}`

And changed the `mypath` stuff below to cover setting the same variable that appears in the expression.

Runtime is down to 30.395s.

LK4D4 commented:
LGTM
